pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -718,7 +718,7 @@ public void initializeDispatchRateLimiterIfNeeded(Optional<Policies> policies) {
 
     @Override
     public boolean trackDelayedDelivery(long ledgerId, long entryId, MessageMetadata msgMetadata) {
-        if (!isDelayedDeliveryEnabled) {
+        if (!isDelayedDeliveryEnabled || !topic.delayedDeliveryEnabled) {
```

Comment:
I think you can remove the isDelayedDeliveryEnabled in dispatcher, maybe we just need it on topic level on subscription level